pgjdbc/src/main/java/org/postgresql/plugin/AuthenticationPlugin.java

```diff
@@ -10,7 +10,6 @@
 import org.checkerframework.checker.nullness.qual.Nullable;
 
 public interface AuthenticationPlugin {
-  @Nullable
-  String getPassword(AuthenticationRequestType type) throws PSQLException;
+  char @Nullable [] getPassword(AuthenticationRequestType type) throws PSQLException;
```

Comment:
We need to document something like "callers should erase the password after use"